### PR TITLE
Fix bug #512: Crash when reading corrupted Jpeg2000 files

### PR DESCRIPTION
**Pull request by pierreldff**

### CODE REVIEW

File: Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx [hunk 1]
```diff
@@ -1687,6 +1693,8 @@ bool JPEG2000Codec::DecodeExtent(
       {
       size_t fraglen = frag.GetVL();
       size_t oldlen = vdummybuffer.size();
+	  if( fraglen == 0 && fraglen == 0 )
```
Comment (malaterre) [inline Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx:1696]: ^ `fraglen` ?

File: Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx [hunk 2]
```diff
@@ -638,7 +638,11 @@ std::pair<char *, size_t> JPEG2000Codec::DecodeByStreamsCommon(char *dummy_buffe
     file_length--;
     }
   // what if 0xd9 is never found ?
-  assert( file_length > 0 && src[file_length-1] == 0xd9 );
+  if( !( file_length > 0 && src[file_length-1] == 0xd9 ) )
+  {
+    assert(true);
```
Comment (malaterre) [inline Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx:643]: remove `assert(true)` thx

Comment (pierreldff) [reply Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx:643]: Ok

File: Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx (hunk 1 above)
Comment (pierreldff) [reply Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx:1696]: Arf ok

**Pull request merged**